### PR TITLE
tests: Add more coverage for dialects/arith

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

**Review by webmiche — APPROVED** [review]
LGTM. I would personally prefer to move the cmath stuff to a differentPR.

File: tests/dialects/test_arith.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                                 Mulf, Divf, Maxf, Minf)
+from xdsl.dialects.builtin import i32, f32
+
+
+class Test_integer_arith_construction:
+    a = Constant.from_int_and_width(1, i32)
+    b = Constant.from_int_and_width(1, i32)
+
+    def test_Addi(self):
+        _ = Addi.get(self.a, self.b)
```
Comment (math-fehr) [inline tests/dialects/test_arith.py:16]: Should we check that both operands are correctly `self.a` and `self.b`?

File: tests/dialects/test_cmath.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Test_cmath:
```
Comment (math-fehr) [inline tests/dialects/test_cmath.py:1]: Should we add this in another PR?
Or should we wait for these changes?

File: xdsl/dialects/cmath.py [hunk 3]
```diff
@@ -40,15 +41,22 @@ class Mul(Operation):
     rhs: Annotated[
         Operand,
         ParamAttrConstraint(ComplexType, [AnyOf([Float32Type, Float64Type])])]
-    res: Annotated[
+    result: Annotated[
```
Comment (math-fehr) [inline xdsl/dialects/cmath.py:44]: Why was this one changed, but not the others?

**Review by math-fehr — CHANGES_REQUESTED** [review]
Thanks! I added some comments, but otherwise looks good!

**Comment by tobiasgrosser** [comment]
Is this something we can push forward. I am trying to clean up the open PRs.

File: tests/dialects/test_arith.py (hunk 1 above)
Comment (georgebisbas) [reply tests/dialects/test_arith.py:16]: Can do

File: tests/dialects/test_cmath.py (hunk 2 above)
Comment (georgebisbas) [reply tests/dialects/test_cmath.py:1]: Can be added in other PR, yes

**Review by math-fehr — APPROVED** [review]
Perfect! Thanks!

**Pull request merged**